Initial implementation of Azure developer signing dataplane service

Jaxelr opened the pull request:
# Data Plane API - Pull Request
This PR is accompanied by the following [related PR](https://github.com/Azure/azure-rest-api-specs/pull/26635). The purpose of this PR is the addition of the newly renamed Azure developer service to the specification. 

This is the initial PR where [Azure Code Signing](https://github.com/Azure/azure-rest-api-specs/pull/22310) was included. 

Also See: https://github.com/Azure/typespec-azure-pr/issues/1999

cc: @DawnWang17, @ashutak84 

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document: [Doc](https://microsoft-my.sharepoint-df.com/:w:/p/rakiasegev/EdK1ryeZ1LNAqwHmrcrQegoB2EZk552BVAo1zjnNiYUBHw?e=zma7Ey&CID=d915a6f2-82ee-7f54-21b2-fe30af4e71ce)
* Previous API Spec Doc: [ApiViewDoc](https://apiview.dev/Assemblies/Review/d470cb25ae414f048c3c58921fdbe83c)
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

Jaxelr committed:
feat: Added optional headers of client version and correlation id

Jaxelr committed:
chore: compiled typespec project

Jaxelr committed:
docs: adding missing readme.md to data-plane spec

Jaxelr committed:
fix: adding words to customwords.txt

mikekistler commented on specification/developersigning/DeveloperSigning/main.tsp:
```diff
@@ -0,0 +1,196 @@
+  sign is StandardResourceOperations.LongRunningResourceAction<
+    CertificateProfileName,
+    SigningPayloadOptions,
+    SignResult,
+    SigningRequestHeadersTraits
```
You are missing the error response here.
```suggestion
    SignResult,
    Foundations.ErrorResponse,
    SigningRequestHeadersTraits
```

Jaxelr replied:
Thanks, this is fixed on the PR.

mikekistler commented on specification/developersigning/DeveloperSigning/main.tsp:
```diff
@@ -0,0 +1,196 @@
+alias SigningRequestHeadersTraits = Azure.Core.Traits.RequestHeadersTrait<{
+  @doc("An optional client version.")
+  @header
+  clientVersion: string;
```
If this header is optional you must indicate that with a "?" after the name.
```suggestion
  clientVersion?: string;
```

Jaxelr replied:
Thanks, this is fixed on the PR.

mikekistler commented on specification/developersigning/DeveloperSigning/main.tsp:
```diff
@@ -0,0 +1,196 @@
+
+  @doc("An identifier used to batch multiple requests.")
+  @header
+  xCorrelationId: string;
```
Is this header also optional?
```suggestion
  xCorrelationId?: string;
```

Jaxelr replied:
Thanks, this is fixed on the PR.

mikekistler reviewed (changes requested):
Just a few little problems. Please address these and then re-request review.

Jaxelr committed:
fix: Mark headers as optional

Jaxelr committed:
fix: Addition of Foundations.ErrorResponse or default errors

Jaxelr committed:
fix: rebase fixes

Jaxelr commented:
Thanks Mike, I just mark the PR to request review.

mikekistler commented on specification/developersigning/DeveloperSigning/tspconfig.yaml:
```diff
@@ -0,0 +1,32 @@
+  "@azure-tools/typespec-autorest":
+    azure-resource-provider-folder: "data-plane"
+    emitter-output-dir: "{project-root}/.."
+    examples-directory: "examples"
```
This will avoid generating unnecessary schemas to your OpenAPI file:
```suggestion
    examples-directory: "examples"
    omit-unreachable-types: true
```

Jaxelr replied:
Great, commited to the branch

mikekistler reviewed (approved):
Looks good. 👍 

I left one more minor comment which will tidy things up but isn't strictly necessary.

Jaxelr committed:
fix: per PR Feedback, modified the tspconfig to purge unreachable types

Jaxelr commented:
@johanste could I get a second review, whenever you have a moment. Thanks.

Jaxelr committed:
fix: examples are not configured for paging

Jaxelr committed:
fix: ran prettier on examples.json files

Jaxelr commented:
Looks like https://github.com/Azure/typespec-azure-pr/issues/3349 is blocking the Swagger linter.

Jaxelr committed:
docs: adding swagger lint suppression

Per feedback from gh issue: https://github.com/Azure/typespec-azure-pr/issues/3349

Jaxelr committed:
feat: initial implementation of azure developer signing service

Jaxelr commented:
hey @anuchandy based on @tjprescott guidance I added a suppression to the swagger linter. Could you please review and let me know what else is needed to Merge this PR?

tjprescott commented:
@jaxelr this needs the sign-off from the API Stewardship board to merge. The Swagger suppression makes sense to me, but they have to approve the in-line suppressions in your TypeSpec.

Jaxelr commented:
@tjprescott I'll post on the Azure API Stewardship teams. Thanks.

mikekistler commented:
API review issue: https://github.com/Azure/azure-rest-api-specs/issues/23716
